### PR TITLE
feat(VTMOutput): Added outputs as .vtm files for vizualisation of mod…

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…els (BRep and Section) in paraview/...

### CODE REVIEW

File: src/geode/io/model/vtm_brep_output.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        VTMBRepOutputImpl( absl::string_view filename, const geode::BRep& brep )
+            : geode::detail::VTMOutputImpl< geode::BRep, 3 >{ filename, brep }
+        {
+        }
+
+    private:
+        void write_piece( pugi::xml_node& object ) final
+        {
+            const auto counter = geode::detail::VTMOutputImpl< geode::BRep,
+                3 >::write_corners_lines_surfaces( object );
```
Comment (BotellaA) [inline src/geode/io/model/vtm_brep_output.cpp:53]:
```suggestion
            const auto counter = this->write_corners_lines_surfaces( object );
```

File: src/geode/io/model/vtm_brep_output.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    else if( const auto* hybrid =
+                                 dynamic_cast< const geode::HybridSolid3D* >(
+                                     &mesh ) )
+                    {
+                        save_hybrid_solid( *hybrid, file );
+                    }
+                    else if( const auto* poly = dynamic_cast<
+                                 const geode::PolyhedralSolid3D* >( &mesh ) )
+                    {
+                        save_polyhedral_solid( *poly, file );
```
Comment (BotellaA) [inline src/geode/io/model/vtm_brep_output.cpp:93]: RegularGrid ?

File: include/geode/io/model/private/vtm_output.h [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                                const TriangulatedSurface< dimension >* >(
+                                &mesh ) )
+                        {
+                            save_triangulated_surface( *triangulated, file );
+                        }
+                        else if( const auto* polygonal = dynamic_cast<
+                                     const PolygonalSurface< dimension >* >(
+                                     &mesh ) )
+                        {
+                            save_polygonal_surface( *polygonal, file );
```
Comment (BotellaA) [inline include/geode/io/model/private/vtm_output.h:174]: RegularGrid ?

File: src/geode/io/model/CMakeLists.txt [hunk 4]
```diff
@@ -26,16 +26,25 @@ add_geode_library(
...
+        "private/vtm_section_output.h"
+        "private/vtm_brep_output.h"
     PRIVATE_DEPENDENCIES
+        Async++
         OpenGeode::basic
         OpenGeode::geometry
         OpenGeode::mesh
         OpenGeode::model
         pugixml::pugixml
+        zlib::zlibstatic
```
[the remark this answers is not in the document]
Comment (MelchiorSchuh) [reply src/geode/io/model/CMakeLists.txt:47]: Yes, since VTKOutputImpl depends on it

**Review by panquez — APPROVED** [review]

File: src/geode/io/model/vtm_section_output.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            : geode::detail::VTMOutputImpl< geode::Section, 2 >{ filename,
+                  section }
+        {
+        }
+
+    private:
+        void write_piece( pugi::xml_node& object ) final
+        {
+            geode::detail::VTMOutputImpl< geode::Section,
+                2 >::write_corners_lines_surfaces( object );
```
Comment (BotellaA) [inline src/geode/io/model/vtm_section_output.cpp:47]: use this also

**Pull request merged**